lib/replicate/checkpointer.js

```diff
@@ -5,24 +5,60 @@ var explain404 = require('./../deps/explain404');
 var pouchCollate = require('pouchdb-collate');
 var collate = pouchCollate.collate;
 
-function updateCheckpoint(db, id, checkpoint, returnValue) {
+var REPLICATION_ID_VERSION = 2;
+// This is an arbitrary number to limit the
+// amount of replication history we save in the checkpoint.
+// If we save too much, the checkpoing docs will become very big,
+// if we save fewer, we'll run a greater risk of having to
+// read all the changes from 0 when checkpoint PUTs fail
+// CouchDB 2.0 has a more involved history pruning,
+// but let's go for the simple version for now.
+var CHECKPOINT_HISTORY_SIZE = 5;
+var LOWEST_SEQ = 0;
+
+function updateCheckpoint(db, id, checkpoint, session, returnValue) {
   return db.get(id).catch(function (err) {
     if (err.status === 404) {
       if (db.type() === 'http') {
         explain404('PouchDB is just checking if a remote checkpoint exists.');
       }
-      return {_id: id};
+      return {
+        session_id: session,
+        _id: id,
+        history: [],
+        replication_id_version: REPLICATION_ID_VERSION
+      };
     }
     throw err;
   }).then(function (doc) {
     if (returnValue.cancelled) {
       return;
     }
+    // Filter out current entry for this replication
+    doc.history = (doc.history || []).filter(function(item) {
```

Comment:
Small nit, we generally put spaces here (need to get a linter to start checking for that)

     doc.history = (doc.history || []).filter(function (item) {